ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/shacl/InMemoryShaclCollection.java

```diff
@@ -1,42 +1,48 @@
 package be.vlaanderen.informatievlaanderen.ldes.server.domain.shacl;
 
 import be.vlaanderen.informatievlaanderen.ldes.server.domain.events.ShaclChangedEvent;
-import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesconfig.services.LdesConfigModelService;
-import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesconfig.valueobjects.LdesConfigModel;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.shacl.entities.ShaclShape;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.shacl.repository.ShaclShapeRepository;
 import jakarta.annotation.PostConstruct;
-import org.apache.jena.rdf.model.Model;
 import org.springframework.context.event.EventListener;
 import org.springframework.stereotype.Component;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
 
 @Component
 public class InMemoryShaclCollection implements ShaclCollection {
-	private final Map<String, Model> shapes;
-	private final LdesConfigModelService ldesConfigModelService;
+	private final Set<ShaclShape> shapes;
+	private final ShaclShapeRepository shaclShapeRepository;
 
-	public InMemoryShaclCollection(LdesConfigModelService ldesConfigModelService) {
-		this.shapes = new HashMap<>();
-		this.ldesConfigModelService = ldesConfigModelService;
+	public InMemoryShaclCollection(ShaclShapeRepository shaclShapeRepository) {
+		this.shaclShapeRepository = shaclShapeRepository;
+		this.shapes = new HashSet<>();
 	}
 
-	@EventListener
-	public void handleShaclChangedEvent(ShaclChangedEvent event) {
-		shapes.put(event.getCollectionName(), event.getShacl());
-		ldesConfigModelService.updateShape(event.getCollectionName(),
-				new LdesConfigModel(event.getCollectionName(), event.getShacl()));
+	@Override
+	public void saveShape(ShaclShape shaclShape) {
+		shaclShapeRepository.saveShaclShape(shaclShape);
+		shapes.remove(shaclShape);
+		shapes.add(shaclShape);
+
 	}
 
 	@Override
-	public LdesConfigModel retrieveShape(String collectionName) {
-		return new LdesConfigModel(collectionName, shapes.get(collectionName));
+	public Optional<ShaclShape> retrieveShape(String collectionName) {
+		return shapes.stream()
+				.filter(shaclShape -> shaclShape.getCollection().equals(collectionName))
+				.findFirst();
+	}
+
+	@EventListener
+	public void handleShaclChangedEvent(ShaclChangedEvent event) {
+		shaclShapeRepository.saveShaclShape(event.getShacl());
+		shapes.remove(event.getShacl());
+		shapes.add(event.getShacl());
```

Comment:
I assume this remove + add works because you only compare the collection in your equality/hashcode of your shaclShape. But if you fix that, this won't work anymore. You can fix this by using a map where the key is the collectionName

Reply:
also just doing 'add' without remove would have the same result as 'remove' + 'add' since duplicates cannot exist in a set and the old one would be replaced